File: app/src/main/java/ezike/tobenna/myweather/data/model/WeatherResponse.java
package ezike.tobenna.myweather.data.model;

import com.squareup.moshi.Json;

import ezike.tobenna.myweather.data.local.entity.CurrentWeather;

/**
 * @author tobennaezike
 */
public class WeatherResponse {

    @Json(name = "current")
    private CurrentWeather current;

    @Json(name = "mWeatherLocation")
    private WeatherLocation mWeatherLocation;

    @Json(name = "forecast")
    private Forecast forecast;

    public CurrentWeather getCurrent() {
        return current;
    }

    public void setCurrent(CurrentWeather current) {
        this.current = current;
    }

    public WeatherLocation getWeatherLocation() {
        return mWeatherLocation;
    }

    public void setWeatherLocation(WeatherLocation weatherLocation) {
        this.mWeatherLocation = weatherLocation;
    }

    public Forecast getForecast() {
        return forecast;
    }

    public void setForecast(Forecast forecast) {
        this.forecast = forecast;
    }
}

File: app/src/main/java/ezike/tobenna/myweather/data/model/WeatherResponse.kt
package ezike.tobenna.myweather.data.model


import androidx.room.Entity
import androidx.room.PrimaryKey
import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
@Entity(tableName = "weather_response")
data class WeatherResponse(
        @PrimaryKey
        val id: Int = 0,
        @Json(name = "current")
        val current: Current,
        @Json(name = "location")
        val weatherLocation: WeatherLocation,
        @Json(name = "request")
        val request: Request
)